daemon/main.go

```diff
@@ -843,6 +843,21 @@ func runDaemon() {
 		// is bootstrapped.
 		d.regenerateRestoredEndpoints(restoredEndpoints)
 		go func() {
+			if k8s.IsEnabled() {
+				log.Debugf("waiting for all services to be listed by Kubernetes before syncing loadbalancer maps")
+				log.Debugf("done waiting for all services to be listed by Kubernetes before syncing loadbalancer maps")
+
+				// Start controller which removes any leftover Kubernetes services
+				// that may have been deleted while Cilium was not running.
+				controller.NewManager().UpdateController("sync-lb-maps-with-k8s-services",
+					controller.ControllerParams{
+						DoFunc: func() error {
+							return d.syncLBMapsWithK8s()
+						},
+					},
+				)
+				return
+			}
```

Comment:
Note that we do not call `SyncLBMap` when running with K8s. I believe that this is fine, primarily because we are trusting K8s to provide us with the information we need instead of the existing state in the datapath; the cleanup happens after we get the services from K8s, so there is no worry about having the maps temporarily cleared of any entry at any given time.